code/game/objects/structures/holosigns.dm

```diff
@@ -62,9 +62,9 @@
 		return TRUE
 	if(mover.pass_flags & (PASSGLASS|PASSTABLE|PASSGRILLE))
 		return TRUE
-	if(iscarbon(mover))
-		var/mob/living/carbon/C = mover
-		if(allow_walk && (C.m_intent == MOVE_INTENT_WALK || (C.pulledby && C.pulledby.m_intent == MOVE_INTENT_WALK)))
+	if(isliving(mover))
+		var/mob/living/walker = mover
+		if(allow_walk && (walker.m_intent == MOVE_INTENT_WALK || (walker.pulledby && walker.pulledby.m_intent == MOVE_INTENT_WALK)))
```

Comment:
To my knowledge, most animals dont actually have an screen object to toggle their walking intent (I'm thinking cats, Ian, terror spiders), that is if they can even toggle their movement intent at all. I think it would be better to have a case  for non-carbons
```
if(!iscarbon(mover)
    return TRUE
    
... rest of code
```

Reply:
fair point, although the default value for m_intent is `MOVE_INTENT_RUN` anyway so this would be functionally the same code. This would also add future support for adding movement intent toggling in the future if it ever happened.